modules/library/referencing/src/main/java/org/geotools/referencing/crs/DefaultProjectedCRS.java

```diff
@@ -299,4 +303,60 @@ protected String formatWKT(final Formatter formatter) {
         formatter.setLinearUnit(linearUnit);
         return "PROJCS";
     }
+
+    @Override
+    public String formatPROJ(PROJFormatter formatter) {
+        final Ellipsoid ellipsoid = ((GeodeticDatum) datum).getEllipsoid();
+        @SuppressWarnings("unchecked") // Formatter.setLinearUnit(...) will do the check for us.
+        final Unit<Length> unit = (Unit) getUnit();
+        final Unit<Length> linearUnit = formatter.getLinearUnit();
+        final Unit<Angle> angularUnit = formatter.getAngularUnit();
+        final Unit<Length> axisUnit = ellipsoid.getAxisUnit();
+        formatter.setLinearUnit(unit);
+        formatter.setAngularUnit(
+                DefaultGeographicCRS.getAngularUnit(baseCRS.getCoordinateSystem()));
+        OperationMethod method = conversionFromBase.getMethod();
+        if (method instanceof PROJFormattable) {
+            formatter.append((PROJFormattable) method);
+        }
+        if (baseCRS instanceof PROJFormattable) formatter.append((PROJFormattable) baseCRS);
+        for (final GeneralParameterValue param : conversionFromBase.getParameterValues().values()) {
+            final GeneralParameterDescriptor desc = param.getDescriptor();
+            String name;
+            if (nameMatches(desc, name = SEMI_MAJOR) || nameMatches(desc, name = SEMI_MINOR)) {
+                /*
+                 * Do not format semi-major and semi-minor axis length in most cases, since this
+                 * information is provided in the ellipsoid. An exception to this rule occurs if
+                 * the lengths are different from the ones declared in the datum.
+                 */
+                if (param instanceof ParameterValue) {
+                    final double value = ((ParameterValue<?>) param).doubleValue(axisUnit);
+                    final double expected =
+                            (SEMI_MINOR.equalsIgnoreCase(name))
+                                    ? // using '==' is okay here.
+                                    ellipsoid.getSemiMinorAxis()
+                                    : ellipsoid.getSemiMajorAxis();
+                    if (value == expected) {
+                        continue;
+                    }
+                }
+            }
+            formatter.append(param);
+        }
+        formatter.append(unit);
+        final int dimension = coordinateSystem.getDimension();
+        if (dimension >= 2) {
+            AxisDirection dir0 = coordinateSystem.getAxis(0).getDirection();
+            AxisDirection dir1 = coordinateSystem.getAxis(1).getDirection();
+            if (dir0 == AxisDirection.WEST && dir1 == AxisDirection.SOUTH) {
+                formatter.append(" +axis=wsu");
+
+            } else if (dir0 == AxisDirection.SOUTH && dir1 == AxisDirection.WEST) {
+                formatter.append(" +axis=swu");
+            }
+        }
+        formatter.setAngularUnit(angularUnit);
+        formatter.setLinearUnit(linearUnit);
```

Comment:
Hasn't this been done at the beginning of the method already? And what's the use of having it at the end?

Reply:
See my COMMENT1 for this

Reply:
unit Logic has been simplified